### PR TITLE
Add measured boot

### PR DESCRIPTION
**Pull request by tomoveu**
This PR adds measured boot by extending the boot image digest calculated by wolfBoot into a TPM PCR. The PCR index is selected by the user.

@danielinux please review and let me know if the way I added the user option is acceptable

@dgarske We should take into account that Microsoft Windows (with the most heavy PCR usage out there) uses PCR0-11 and the only remaining **safe PCRs** for use by us, as a bootloader, are PCR12-15.

### CODE REVIEW

File: src/image.c
Comment (dgarske) [inline src/image.c:678]: Consider putting this after the memcmp, so its only updated on success.

Comment (tomoveu) [reply src/image.c:678]: I had this comment in my initial code, but removed it in the last moment. Perhaps I should re-add it?
+    /*
+     * TPM Measurement must be performed regardless of the
+     * verification outcome afterwards, becuse the purpose
+     * of a Measured Boot is to record the current boot state
+     */

File: options.mk
Comment (dgarske) [inline options.mk:190]: Please add this new option `MEASURED_BOOT` to the `tools/config.mk`. Add documentation for the new option. Note it requires `.config` with `WOLFTPM=1` and `MEASURED_BOOT=[PCRNUM]`.

Comment (tomoveu) [reply options.mk:190]: I was wondering if I could code this requirement the way we do in wolfTPM, but it seems with option.mk such dependencies are not possible. I will make sure to add detailed info in documentation. Thanks

Comment (dgarske) [reply options.mk:190]: Yes, code it to force WOLFTPM=1 in the Makefile

File: src/image.c
Comment (dgarske) [reply src/image.c:678]: Yes add this comment.

**Comment by tomoveu** [comment]
Thanks for the review @dgarske

Both suggestions are added and I want to finalize the testing before making the new push.

File: options.mk
Comment (danielinux) [reply options.mk:190]: or even better, in `options.mk`

**Comment by tomoveu** [comment]
Feedback taken in. Some testing done too. I will add updated _README.md_ and new file in _docs/measured_boot.md_ to finalize this PR. 

And I want to test bit more the code, because I had some issues with my HW setup .Purely HW stuff, SPI wiring issue between my STM32F4 and LetsTrust TPM.

Also, @danielinux  made a great suggestion for an extra effort - To add an example of measured_boot app in the _wolfboot-examples_ repo. This would be a separate effort and if time allows it. So, right now I have to push the docs and do a bit more testing.

**Comment by tomoveu** [comment]
New feature tested with Infineon TPM9670_Raspberry module and seems to be working as expected.

I verified that PRC extend operations, using the same firmware image, are reproducible in wolfBoot :)

@danielinux please review the code, I will wrap this PR with updating the documentation as discussed.

**Comment by tomoveu** [comment]
Documentation about measured boot added. README.md is also updated.

I would like at least one more person to test on the real hardware before I merge this PR.

I am making a new wolfBoot-example for measured boot, but that would be a PR for the other repo.

**Review by dgarske — APPROVED** [review]

**Comment by dgarske** [comment]
@tomoveu : I will hold off to merge until I can setup on hardware here with your wolfboot-examples repo (once pushed).

**Comment by tomoveu** [comment]
@dgarske Just to make sure there is no misunderstanding: This is the measured boot implementation and it is ready for merging.

It can be tested by using gdb, and checking the PCR value is reproducible when using the same test-app (default one works).

The new wolfboot-example is only to make it easier for people to use the measured boot. It is not required for having measured boot.

**Pull request merged**
